src/data/fin.lean

```diff
@@ -196,6 +196,10 @@ def coe_embedding (n) : (fin n) ↪o ℕ :=
 instance fin.lt.is_well_order (n) : is_well_order (fin n) (<) :=
 (coe_embedding n).is_well_order
 
+/-- Use the ordering on `fin n` for checking recursive definitions. -/
+instance {n : ℕ} : has_well_founded (fin n) :=
+⟨_, measure_wf coe⟩
```

Comment:
You no longer use this in this PR, right? Would you mind adding an example of what this enables in its docstring?